### PR TITLE
Add non-root EFS documentation

### PR DESCRIPTION
**Pull request by droctothorpe**
Co-authored-by: travischeung <travis.cheung98@gmail.com>

This PR improves the JupyterHub EFS integration documentation and adds instructions for how to mount EFS volumes to your home directory _as non-root_. This is made possible by the  [aws-efs-csi-driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver) and EFS [Access
Points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html).

We also enforced text wrapping in the document as a whole, hence the appearance of more changes than there actually are.

### CODE REVIEW

File: jupyterhub/schema.yaml [hunk 1]
```diff
@@ -429,7 +429,7 @@ properties:
           some custom way.
 
           For more details, see the [Kubernetes
-          documentation](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes).
```
Comment (droctothorpe) [inline jupyterhub/schema.yaml:432]: The `#notes` anchor was removed from the hyperlink. This was leading to a link check failure in CI, so we removed the anchor.

File: jupyterhub/schema.yaml [hunk 2]
```diff
@@ -472,7 +472,7 @@ properties:
           ```
 
           For more details, see the [Kubernetes
-          documentation](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes).
```
Comment (droctothorpe) [inline jupyterhub/schema.yaml:475]: The `#notes` anchor was removed from the hyperlink. This was leading to a link check failure in CI, so we removed the anchor.

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/mounting-efs-nfs-to-home-without-root-privileges/9837/6

File: docs/source/kubernetes/amazon/efs_storage.md [hunk 3]
```diff
@@ -4,132 +4,269 @@ orphan: true
...
+Persistent Volume Claim to your Kubernetes cluster:
+
+```bash
+kubectl create namespace <your namespace>
+kubectl apply --namespace=<your namespace> -f sc.yaml -f pv.yaml -f pvc.yaml
+```
+
+### Configuring JupyterHub to use the EFS PV/PVC (non-root)
+
+Add the follow helm chart:
```
Comment (droctothorpe) [inline docs/source/kubernetes/amazon/efs_storage.md:252]:
```suggestion
Add the following to
[values.yaml](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/jupyterhub/values.yaml#L360)
in the JupyterHub helm chart:
```

**Comment by droctothorpe** [comment]
Squashed commits for a simplified merge.

**Comment by droctothorpe** [comment]
Checking in on this. More than happy to address any feedback.

**Comment by droctothorpe** [comment]
Resolved conflicts with main. Standing by for a response. 

🦗 😂

**Review by consideRatio — COMMENTED** [review]
Hmmm @droctothorpe can you undo the changes that are just indentation changes so I can review what you actually have changed more easily?

If you would like to include that kind of a change in a PR, you could make it a dedicated commit and ask that the reviewer disregard reviewing the changes in that commit specifically and that it makes sense to review commit by commit due to that so that one can focus on that content changes.

**Comment by consideRatio** [comment]
@droctothorpe I'll go for a close as this went stale, but if you want to work it I'll commit to reviewing it!

**Comment by droctothorpe** [comment]
Hey, Erik. No worries. I appreciate the offer to revisit. Just went through a pretty substantial re-org and tackling a completely unrelated workstream, so haven't had a chance to loop back around to this. It would be a shame for it to slip through the cracks though because running user containers as non-root is a firm requirement for a lot of companies / cluster-admins. I'll try to get back around to it. Leaving it closed for now is perfectly reasonable.

**Pull request closed without merging**